src/Validator.php

```diff
@@ -235,19 +233,37 @@ public static function validateType($typeName)
      * Validates a type to be used as a method parameter type hint.
      *
      * @param string $typeName The name of the type to test.
+     * @param bool $allowClass Allow class type hints or not
      * @return null|string Returns a valid type hint for the type or null if there is no valid type hint.
      */
-    public static function validateTypeHint($typeName)
+    public static function validateTypeHint($typeName, $allowClass = false)
     {
-        $typeHint = null;
+        if (self::isInternalPhpType($typeName)) {
+            return null;
+        }
+
+        //TODO: scalar type hints are avalible in PHP7 only
+        if (self::isInternalPhpType($typeName)) {
+            return null;
+        }
 
         // We currently only support type hints for arrays and DateTimes.
         // Going forward we could support it for generated types. The challenge here are enums as they are actually
         // strings and not class instances and we have no way of determining whether the type is an enum at this point.
         if (substr($typeName, -2) == "[]") {
-            $typeHint = 'array';
+            return 'array';
```

Comment:
Would love to discuss Enums. It has been a reoccurring challenge regarding type hinting. The issue was raised a while ago in #91.